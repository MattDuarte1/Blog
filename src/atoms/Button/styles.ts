import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme, disabled }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 20px 23px 19px;
    gap: 10px;
    border: none;
    cursor: ${disabled ? 'not-allowed' : 'pointer'};

    min-width: 145px;
    height: 67px;
    font-weight: 700;
    color: #ffff;

    background: ${theme.colors.blue};
    border-radius: 8px;
    transition: filter 0.4s ease;

    filter: ${disabled && 'brightness(0.7)'};

    &:hover {
      filter: ${!disabled && 'brightness(1.1)'};
    }
  `}
`;
