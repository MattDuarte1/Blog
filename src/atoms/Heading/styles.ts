import styled, { css, DefaultTheme } from 'styled-components';

interface ContainerProps {
  size?:
    | 'xxsmall'
    | 'xsmall'
    | 'small'
    | 'medium'
    | 'large'
    | 'xlarge'
    | 'xxlarge'
    | 'huge'
    | 'xhuge';
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue';
  uppercase: boolean;
}
const MediaFont = {
  large: (theme: DefaultTheme) => css`
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.medium};
    }
  `,
  xxlarge: (theme: DefaultTheme) => css`
    @media ${theme.media.lteMedium} {
      font-size: ${theme.font.sizes.xlarge};
    }
    @media ${theme.media.lteSmall} {
      font-size: ${theme.font.sizes.large};
    }
  `,
};

const TitleSize = {
  xxsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxsmall};
  `,
  xsmall: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xsmall};
  `,
  small: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.small};
  `,
  medium: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.medium};
  `,
  large: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.large};
    ${MediaFont.large(theme)};
  `,
  xlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,
  xxlarge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xxlarge};
    ${MediaFont.xxlarge(theme)};
  `,
  huge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.huge};
  `,
  xhuge: (theme: DefaultTheme) => css`
    font-size: ${theme.font.sizes.xhuge};
  `,
};

const TitleCase = (uppercase: boolean) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

const TitleColor = (
  color:
    | 'white'
    | 'lightGrey1'
    | 'lightGrey2'
    | 'grey'
    | 'darkGrey'
    | 'black'
    | 'blue',
  theme: DefaultTheme,
) => css`
  color: ${color ? theme.colors[color] : theme.colors.black};
`;

export const Container = styled.h1<ContainerProps>`
  ${({ theme, size, color, uppercase }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 700;
    line-height: 1.2;
    text-transform: capitalize;

    ${TitleSize[size](theme)}
    ${TitleCase(uppercase)}
    ${TitleColor(color, theme)}
  `}
`;
