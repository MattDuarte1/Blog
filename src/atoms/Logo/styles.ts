import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.font.family.default};
    font-style: normal;
    font-weight: 600;
    font-size: ${theme.font.sizes.large};
    line-height: 45px;
    position: relative;
    &::first-letter {
      color: ${theme.colors.blue};
    }

    &::after {
      content: '.blog';
      font-family: ${theme.font.family.default};
      font-style: normal;
      font-weight: 600;
      font-size: ${theme.font.sizes.small};
      color: ${theme.colors.blue};
    }
  `}
`;
