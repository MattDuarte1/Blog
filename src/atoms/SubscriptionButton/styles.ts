import styled, { css } from 'styled-components';

export const Container = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.blue};
    border: none;
    border-radius: 25px;
    display: flex;
    justify-content: center;
    align-items: center;
    gap: 0.9rem;
    padding: 1rem;

    cursor: pointer;
  `}
`;
