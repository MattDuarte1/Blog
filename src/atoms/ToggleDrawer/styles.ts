import styled, { css } from 'styled-components';

type IToggle = {
  active?: boolean;
};

export const Toggle = styled.button<IToggle>`
  ${({ theme, active }) => css`
    background-color: #fff;
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.08);
    border-radius: 10px;
    border: none;
    overflow: hidden;
    display: none;
    cursor: pointer;
    justify-content: center;
    align-items: center;
    position: relative;
    height: 70px;
    width: 70px;

    p {
      position: absolute;
      width: 40px;
      height: 4px;
      background-color: ${theme.colors.blue};
      border-radius: 4px;
      transition: 0.5s;
    }

    p:nth-child(1) {
      transform: ${active
        ? 'translateY(0px) rotate(45deg)'
        : 'translateY(-15px)'};
      width: ${active ? '40px' : '25px'};
      left: 15px;
    }

    p:nth-child(2) {
      transform: ${active
        ? 'translateY(0px) rotate(315deg)'
        : 'translateY(15px)'};
      width: ${active ? '40px' : '15px'};
      left: 15px;
    }

    p:nth-child(3) {
      transform: ${active && 'translateX(60px)'};
    }

    @media ${theme.media.lteLarge} {
      display: flex;
    }
  `}
`;
