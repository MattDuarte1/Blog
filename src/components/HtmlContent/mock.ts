export default '<p>Se voc&ecirc; deseja se tornar um desenvolvedor web para poder desenvolver p&aacute;ginas completas, com layouts elegantes e conte&uacute;do din&acirc;mico, voc&ecirc; precisa conhecer o JavaScript.</p>\n<p>O JavaScript &eacute; a terceira etapa para construirmos um site profissional. No post anterior conhecemos as duas primeiras etapas:<span>&nbsp;</span><a href="https://www.treinaweb.com.br/blog/como-comecar-com-html-e-css/" title="o HTML e o CSS">o HTML e o CSS</a>, e nesse artigo vamos conhecer sobre como deixar o conte&uacute;do de nossas p&aacute;ginas mais interativo utilizando o JavaScript.</p>\n<h2>O que &eacute; JavaScript?</h2>\n<p>Antes de come&ccedil;armos a abordar o que vem a ser o JavaScript, &eacute; muito importante ressaltar que JavaScript n&atilde;o tem nada a ver com Java.</p>\n<p>Quando essa linguagem foi lan&ccedil;ada, ela tinha o nome oficial de LiveScript, criada pela Netscape. Como nessa &eacute;poca o Java estava conquistando cada vez mais o mercado, a Netscape quis aproveitar a popularidade do rec&eacute;m-lan&ccedil;ado Java, atrav&eacute;s de um acordo com a Sun, mudando ent&atilde;o seu nome para JavaScript.</p>\n<p>Essa mudan&ccedil;a ainda causa confus&atilde;o dando a impress&atilde;o de que a linguagem foi baseada em Java, mas n&atilde;o&hellip; foi tudo uma estrat&eacute;gia de marketing. Agora sim, podemos conhecer o que realmente vem a ser o JavaScript.</p>\n<p>O JavaScript &eacute; uma linguagem de programa&ccedil;&atilde;o interpretada, sendo a principal linguagem cliente-servidor em navegadores web. Ela &eacute; respons&aacute;vel pelo dinamismo de uma p&aacute;gina, ou seja, promover a intera&ccedil;&atilde;o com o usu&aacute;rio, sendo executados do lado do cliente. &Eacute; uma linguagem script com orienta&ccedil;&atilde;o a objetos din&acirc;mica e com tipagem fraca.</p>\n<p>Existem tr&ecirc;s camadas b&aacute;sicas no desenvolvimento web: o HTML e o CSS (como vimos no post anterior) e por fim a terceira camada que &eacute; o JavaScript, que como vimos acima, fica com a parte de comportamento da p&aacute;gina.</p>\n<h2>Quem usa JavaScript?</h2>\n<p>V&aacute;rias empresas e sites conhecidos utilizam JavaScript na parte de front-end, desde aplica&ccedil;&otilde;es mais simples como as mais complexas. Temos como exemplo a Google, Youtube, Facebook, Ebay, Yahoo, Twitter, Linkedin, dentre muitos outros. Quase toda aplica&ccedil;&atilde;o Web tem a necessidade de usar JavaScript. E, hoje em dia, o JavaScript n&atilde;o se limita apenas &agrave; web, ele tem sua vertente server-side, mobile etc.</p>\n<h2>Sintaxe do JavaScript</h2>\n<p>A sintaxe do JavaScript &eacute; muito simples. Voc&ecirc; pode utilizar o JavaScript dentro do seu c&oacute;digo HTML. Para criarmos o nosso cl&aacute;ssico &ldquo;Hello World&rdquo;, &eacute; s&oacute; colocamos a escrita dentro de um &ldquo;alert&rdquo;. Com o alert nosso texto aparecer&aacute; dentro de uma janela de alerta.</p>\n<p><button class="copy-code-button" type="button">Copiar</button></p>\n<pre class=" language-html" tabindex="0"><code class=" language-html"><span class="token tag"><span class="token punctuation">&lt;</span>script <span class="token attr-name">type</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">"</span>text/javascript<span class="token punctuation">"</span></span><span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">\n       <span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'Hello World!\'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>\n</span></span><span class="token tag"><span class="token punctuation">&lt;/</span>script<span class="token punctuation">&gt;</span></span>\n</code></pre>\n<p>Realizar uma estrutura de condi&ccedil;&atilde;o no JavaScript tamb&eacute;m &eacute; simples:</p>\n<p><button class="copy-code-button" type="button">Copiar</button></p>\n<pre class=" language-html" tabindex="0"><code class=" language-html"><span class="token tag"><span class="token punctuation">&lt;</span>script<span class="token punctuation">&gt;</span></span><span class="token script"><span class="token language-javascript">\n    <span class="token keyword">var</span> curso <span class="token operator">=</span> <span class="token string">\'javascript\'</span><span class="token punctuation">;</span>\n    <span class="token keyword control-flow">if</span> <span class="token punctuation">(</span>curso <span class="token operator">===</span> <span class="token string">\'javascript\'</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>\n        <span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'Oba! Ent&atilde;o inicie agora mesmo seus estudos em JavaScript.\'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>\n    <span class="token punctuation">}</span> <span class="token keyword control-flow">else</span> <span class="token punctuation">{</span>\n        <span class="token function">alert</span><span class="token punctuation">(</span><span class="token string">\'Ah! Parece que JavaScript n&atilde;o &eacute; sua linguagem favorita...\'</span><span class="token punctuation">)</span><span class="token punctuation">;</span>\n    <span class="token punctuation">}</span>\n</span></span><span class="token tag"><span class="token punctuation">&lt;/</span>script<span class="token punctuation">&gt;</span></span>\n</code></pre>\n<p>No JavaScript voc&ecirc; tamb&eacute;m pode definir suas pr&oacute;prias fun&ccedil;&otilde;es. Vamos ent&atilde;o escrever uma fun&ccedil;&atilde;o que recebe dois n&uacute;meros e realiza a opera&ccedil;&atilde;o de multiplica&ccedil;&atilde;o:</p>\n<p><button class="copy-code-button" type="button">Copiar</button></p>\n<pre class=" language-javascript" tabindex="0"><code class=" language-javascript"><span class="token keyword">function</span> <span class="token function">multiplicacao</span><span class="token punctuation">(</span><span class="token parameter">num1<span class="token punctuation">,</span>num2</span><span class="token punctuation">)</span> <span class="token punctuation">{</span>\n     <span class="token keyword">var</span> resultado <span class="token operator">=</span> num1 <span class="token operator">*</span> num2<span class="token punctuation">;</span>\n     <span class="token keyword control-flow">return</span> resultado<span class="token punctuation">;</span>\n<span class="token punctuation">}</span>\n</code></pre>\n<h2>Mercado de trabalho</h2>\n<p>Como pudemos ver o JavaScript &eacute; uma das etapas do desenvolvimento web. Com isso, &eacute; obrigat&oacute;rio um desenvolvedor conhecer essa linguagem. Existem v&aacute;rias bibliotecas para se trabalhar com JavaScript, por isso &eacute; importante ter conhecimento de algumas delas, pois &eacute; muito comum elas serem requisitos para o preenchimento de vagas JavaScript.</p>\n<p>Al&eacute;m da adequa&ccedil;&atilde;o ao mercado de trabalho, conhecer essas bibliotecas garante um aumento de produtividade e facilidade em tarefas comuns do dia a dia.</p>\n<p>Sendo assim, como todos os sites possuem algum tipo de intera&ccedil;&atilde;o utilizando o JavaScript, dificilmente voc&ecirc; ficar&aacute; fora do mercado.</p>\n<h2>Como aprender JavaScript?</h2>\n<p>Aqui no TreinaWeb temos toda uma trilha voltada para o JavaScript. Para voc&ecirc; come&ccedil;ar nessa linguagem, recomendamos o curso &ldquo;JavaScript B&aacute;sico&rdquo;.</p>\n<p>Com ele, voc&ecirc; aprender&aacute; desde os conceitos b&aacute;sicos, fazendo com que voc&ecirc; entenda toda a estrutura da linguagem.</p>\n<p>Aliando a teoria com a pr&aacute;tica, voc&ecirc; tamb&eacute;m ir&aacute; desenvolver programas que ir&atilde;o interagir com os elementos de uma p&aacute;gina web.</p>\n<h2>Conhe&ccedil;a nossa metodologia</h2>\n<p>Nosso ambiente de aprendizagem &eacute; 100% online. Nossa metodologia de ensino &eacute; baseada em v&iacute;deo aulas, apostilas, exerc&iacute;cios aula-a-aula e exemplos completos. Voc&ecirc; estuda no seu tempo e ao final pode baixar o seu certificado digital ou solicitar o certificado impresso.</p>\n<p>Na parte escrita, fornecemos um material totalmente atualizado e completo para que voc&ecirc; possa usufruir do conte&uacute;do, utilizando-se de textos e imagens. J&aacute; as v&iacute;deo-aulas s&atilde;o bem explicativas e pr&aacute;ticas, abordando todo o conte&uacute;do passo-a-passo, para que voc&ecirc; entenda tudo da melhor maneira poss&iacute;vel.</p>\n<p>Al&eacute;m disso, voc&ecirc; encontrar&aacute; exerc&iacute;cios para poder fixar, praticar e aplicar todo o conte&uacute;do aprendido. S&atilde;o exerc&iacute;cios que s&atilde;o propostos em cada aula do curso. Eles podem variar entre alguns formatos, como por exemplo quest&otilde;es de escolha &uacute;nica, quest&otilde;es de m&uacute;ltipla escolha e complemento de trechos de c&oacute;digo.</p>\n<p>Se ainda assim voc&ecirc; tiver alguma d&uacute;vida, voc&ecirc; poder&aacute; pedir ajuda aos nossos professores atrav&eacute;s de tickets de atendimento. Funciona como um canal direto entre voc&ecirc; e o professor, onde voc&ecirc; poder&aacute; realizar perguntas que ser&atilde;o respondidas muito brevemente!</p>\n<p>Al&eacute;m disso, ainda pelo sistema suporte, voc&ecirc; tamb&eacute;m poder&aacute; consultar todas as perguntas que j&aacute; foram respondidas a outros alunos! =)</p>\n<h2>Quais conhecimentos preciso ter para iniciar este curso?</h2>\n<p>O curso &eacute; recomendado para voc&ecirc; iniciar seus estudos na linguagem JavaScript, no entanto, &eacute; interessante saber as bases necess&aacute;rias para aprender qualquer linguagem de programa&ccedil;&atilde;o: &ldquo;L&oacute;gica de Programa&ccedil;&atilde;o&rdquo; e &ldquo;L&oacute;gica de Programa&ccedil;&atilde;o Orientada a Objetos&rdquo;.</p>\n<h2>Gostou do que viu?</h2>\n<p>Se entusiasmou com o JavaScript? Gostaria de conhec&ecirc;-lo com mais detalhes? Sim, voc&ecirc; pode dar uma espiadinha. Acesse o curso de JavaScript e veja algumas aulas de exemplo. :)</p>\n<p>Bons estudos!</p>';
