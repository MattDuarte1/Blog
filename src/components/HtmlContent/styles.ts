import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.white};
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    padding: 10rem;
    max-width: 100%;
    line-height: 1.5;

    iframe {
      width: 500px;
    }

    button {
      background-color: ${theme.colors.lightGrey2};
      padding: ${theme.spacings.xsmall};
      cursor: pointer;
      border: none;
    }

    p {
      margin: ${theme.spacings.medium} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.grey};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }
    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colors.black};
      font-size: ${theme.font.sizes.xsmall};
      background: ${theme.colors.darkGrey};
      padding: 0.2rem;
      margin: 0.2rem;
    }
    pre {
      background: ${theme.colors.darkGrey};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.black};
      margin: ${theme.spacings.medium} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
      height: auto;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.lightGrey1};
    }
    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 90%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
      background-color: ${theme.colors.darkGrey};
    }
    table td,
    table th {
      padding: ${theme.spacings.xsmall};
      border: 0.1rem solid ${theme.colors.lightGrey1};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.lightGrey1};
      color: ${theme.colors.darkGrey};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }
    @media ${theme.media.lteMedium} {
      font-size: 2rem;
      padding: 3rem;

      iframe {
        width: 100%;
      }
    }
  `}
`;
