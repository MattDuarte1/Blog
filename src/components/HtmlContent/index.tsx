import * as Styled from './styles';

type HtmlContentProps = {
  html: string;
};

const HtmlContent = ({ html }: HtmlContentProps) => {
  return <Styled.Container dangerouslySetInnerHTML={{ __html: html }} />;
};

export default HtmlContent;
