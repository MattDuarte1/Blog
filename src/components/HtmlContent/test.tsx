import HtmlContent from '.';
import { renderTheme } from '../../styles/render-theme';
import htmlMocked from './mock';

describe('<HtmlContent>', () => {
  it('Should render a HtmlContent', () => {
    const { container } = renderTheme(<HtmlContent html={htmlMocked} />);

    expect(container).toBeInTheDocument();
  });

  it('Should match snapshot component', () => {
    const { container } = renderTheme(<HtmlContent html={htmlMocked} />);

    expect(container).toMatchSnapshot();
  });
});
