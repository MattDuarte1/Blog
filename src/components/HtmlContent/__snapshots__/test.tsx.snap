// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HtmlContent> Should match snapshot component 1`] = `
.c0 {
  background-color: #FFFFFE;
  font-size: calc(2.156rem + 0.2rem);
  padding: 10rem;
  max-width: 100%;
  line-height: 1.5;
}

.c0 iframe {
  width: 500px;
}

.c0 button {
  background-color: #E4E5E9;
  padding: 1.617rem;
  cursor: pointer;
  border: none;
}

.c0 p {
  margin: 2.4rem 0;
}

.c0 a,
.c0 a:visited,
.c0 a:link {
  color: #C0C0C0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c0 code {
  font-family: monospace;
  color: #2B2C34;
  font-size: 1.617rem;
  background: #9A9494;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c0 pre {
  background: #9A9494;
  padding: 2.4rem;
  font-family: monospace;
  color: #2B2C34;
  margin: 2.4rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 2.156rem;
}

.c0 pre code {
  color: inherit;
  background: inherit;
}

.c0 img {
  max-width: 100%;
  height: auto;
}

.c0 hr {
  border: none;
  border-bottom: 0.1rem solid #EFF0F3;
}

.c0 ul,
.c0 ol {
  margin: 2.4rem 4.8rem;
}

.c0 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c0 table {
  width: 90%;
  border-collapse: collapse;
  margin: 2.4rem 0;
  background-color: #9A9494;
}

.c0 table td,
.c0 table th {
  padding: 1.617rem;
  border: 0.1rem solid #EFF0F3;
}

.c0 blockquote {
  border-left: 0.5rem solid #EFF0F3;
  color: #9A9494;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

@media (max-width:700px) {
  .c0 {
    font-size: 2rem;
    padding: 3rem;
  }

  .c0 iframe {
    width: 100%;
  }
}

<div>
  <div
    class="c0"
  >
    <p>
      Se você deseja se tornar um desenvolvedor web para poder desenvolver páginas completas, com layouts elegantes e conteúdo dinâmico, você precisa conhecer o JavaScript.
    </p>
    

    <p>
      O JavaScript é a terceira etapa para construirmos um site profissional. No post anterior conhecemos as duas primeiras etapas:
      <span>
         
      </span>
      <a
        href="https://www.treinaweb.com.br/blog/como-comecar-com-html-e-css/"
        title="o HTML e o CSS"
      >
        o HTML e o CSS
      </a>
      , e nesse artigo vamos conhecer sobre como deixar o conteúdo de nossas páginas mais interativo utilizando o JavaScript.
    </p>
    

    <h2>
      O que é JavaScript?
    </h2>
    

    <p>
      Antes de começarmos a abordar o que vem a ser o JavaScript, é muito importante ressaltar que JavaScript não tem nada a ver com Java.
    </p>
    

    <p>
      Quando essa linguagem foi lançada, ela tinha o nome oficial de LiveScript, criada pela Netscape. Como nessa época o Java estava conquistando cada vez mais o mercado, a Netscape quis aproveitar a popularidade do recém-lançado Java, através de um acordo com a Sun, mudando então seu nome para JavaScript.
    </p>
    

    <p>
      Essa mudança ainda causa confusão dando a impressão de que a linguagem foi baseada em Java, mas não… foi tudo uma estratégia de marketing. Agora sim, podemos conhecer o que realmente vem a ser o JavaScript.
    </p>
    

    <p>
      O JavaScript é uma linguagem de programação interpretada, sendo a principal linguagem cliente-servidor em navegadores web. Ela é responsável pelo dinamismo de uma página, ou seja, promover a interação com o usuário, sendo executados do lado do cliente. É uma linguagem script com orientação a objetos dinâmica e com tipagem fraca.
    </p>
    

    <p>
      Existem três camadas básicas no desenvolvimento web: o HTML e o CSS (como vimos no post anterior) e por fim a terceira camada que é o JavaScript, que como vimos acima, fica com a parte de comportamento da página.
    </p>
    

    <h2>
      Quem usa JavaScript?
    </h2>
    

    <p>
      Várias empresas e sites conhecidos utilizam JavaScript na parte de front-end, desde aplicações mais simples como as mais complexas. Temos como exemplo a Google, Youtube, Facebook, Ebay, Yahoo, Twitter, Linkedin, dentre muitos outros. Quase toda aplicação Web tem a necessidade de usar JavaScript. E, hoje em dia, o JavaScript não se limita apenas à web, ele tem sua vertente server-side, mobile etc.
    </p>
    

    <h2>
      Sintaxe do JavaScript
    </h2>
    

    <p>
      A sintaxe do JavaScript é muito simples. Você pode utilizar o JavaScript dentro do seu código HTML. Para criarmos o nosso clássico “Hello World”, é só colocamos a escrita dentro de um “alert”. Com o alert nosso texto aparecerá dentro de uma janela de alerta.
    </p>
    

    <p>
      <button
        class="copy-code-button"
        type="button"
      >
        Copiar
      </button>
    </p>
    

    <pre
      class=" language-html"
      tabindex="0"
    >
      <code
        class=" language-html"
      >
        <span
          class="token tag"
        >
          <span
            class="token punctuation"
          >
            &lt;
          </span>
          script 
          <span
            class="token attr-name"
          >
            type
          </span>
          <span
            class="token attr-value"
          >
            <span
              class="token punctuation attr-equals"
            >
              =
            </span>
            <span
              class="token punctuation"
            >
              "
            </span>
            text/javascript
            <span
              class="token punctuation"
            >
              "
            </span>
          </span>
          <span
            class="token punctuation"
          >
            &gt;
          </span>
        </span>
        <span
          class="token script"
        >
          <span
            class="token language-javascript"
          >
            
       
            <span
              class="token function"
            >
              alert
            </span>
            <span
              class="token punctuation"
            >
              (
            </span>
            <span
              class="token string"
            >
              'Hello World!'
            </span>
            <span
              class="token punctuation"
            >
              )
            </span>
            <span
              class="token punctuation"
            >
              ;
            </span>
            

          </span>
        </span>
        <span
          class="token tag"
        >
          <span
            class="token punctuation"
          >
            &lt;/
          </span>
          script
          <span
            class="token punctuation"
          >
            &gt;
          </span>
        </span>
        

      </code>
    </pre>
    

    <p>
      Realizar uma estrutura de condição no JavaScript também é simples:
    </p>
    

    <p>
      <button
        class="copy-code-button"
        type="button"
      >
        Copiar
      </button>
    </p>
    

    <pre
      class=" language-html"
      tabindex="0"
    >
      <code
        class=" language-html"
      >
        <span
          class="token tag"
        >
          <span
            class="token punctuation"
          >
            &lt;
          </span>
          script
          <span
            class="token punctuation"
          >
            &gt;
          </span>
        </span>
        <span
          class="token script"
        >
          <span
            class="token language-javascript"
          >
            
    
            <span
              class="token keyword"
            >
              var
            </span>
             curso 
            <span
              class="token operator"
            >
              =
            </span>
             
            <span
              class="token string"
            >
              'javascript'
            </span>
            <span
              class="token punctuation"
            >
              ;
            </span>
            
    
            <span
              class="token keyword control-flow"
            >
              if
            </span>
             
            <span
              class="token punctuation"
            >
              (
            </span>
            curso 
            <span
              class="token operator"
            >
              ===
            </span>
             
            <span
              class="token string"
            >
              'javascript'
            </span>
            <span
              class="token punctuation"
            >
              )
            </span>
             
            <span
              class="token punctuation"
            >
              {
            </span>
            
        
            <span
              class="token function"
            >
              alert
            </span>
            <span
              class="token punctuation"
            >
              (
            </span>
            <span
              class="token string"
            >
              'Oba! Então inicie agora mesmo seus estudos em JavaScript.'
            </span>
            <span
              class="token punctuation"
            >
              )
            </span>
            <span
              class="token punctuation"
            >
              ;
            </span>
            
    
            <span
              class="token punctuation"
            >
              }
            </span>
             
            <span
              class="token keyword control-flow"
            >
              else
            </span>
             
            <span
              class="token punctuation"
            >
              {
            </span>
            
        
            <span
              class="token function"
            >
              alert
            </span>
            <span
              class="token punctuation"
            >
              (
            </span>
            <span
              class="token string"
            >
              'Ah! Parece que JavaScript não é sua linguagem favorita...'
            </span>
            <span
              class="token punctuation"
            >
              )
            </span>
            <span
              class="token punctuation"
            >
              ;
            </span>
            
    
            <span
              class="token punctuation"
            >
              }
            </span>
            

          </span>
        </span>
        <span
          class="token tag"
        >
          <span
            class="token punctuation"
          >
            &lt;/
          </span>
          script
          <span
            class="token punctuation"
          >
            &gt;
          </span>
        </span>
        

      </code>
    </pre>
    

    <p>
      No JavaScript você também pode definir suas próprias funções. Vamos então escrever uma função que recebe dois números e realiza a operação de multiplicação:
    </p>
    

    <p>
      <button
        class="copy-code-button"
        type="button"
      >
        Copiar
      </button>
    </p>
    

    <pre
      class=" language-javascript"
      tabindex="0"
    >
      <code
        class=" language-javascript"
      >
        <span
          class="token keyword"
        >
          function
        </span>
         
        <span
          class="token function"
        >
          multiplicacao
        </span>
        <span
          class="token punctuation"
        >
          (
        </span>
        <span
          class="token parameter"
        >
          num1
          <span
            class="token punctuation"
          >
            ,
          </span>
          num2
        </span>
        <span
          class="token punctuation"
        >
          )
        </span>
         
        <span
          class="token punctuation"
        >
          {
        </span>
        
     
        <span
          class="token keyword"
        >
          var
        </span>
         resultado 
        <span
          class="token operator"
        >
          =
        </span>
         num1 
        <span
          class="token operator"
        >
          *
        </span>
         num2
        <span
          class="token punctuation"
        >
          ;
        </span>
        
     
        <span
          class="token keyword control-flow"
        >
          return
        </span>
         resultado
        <span
          class="token punctuation"
        >
          ;
        </span>
        

        <span
          class="token punctuation"
        >
          }
        </span>
        

      </code>
    </pre>
    

    <h2>
      Mercado de trabalho
    </h2>
    

    <p>
      Como pudemos ver o JavaScript é uma das etapas do desenvolvimento web. Com isso, é obrigatório um desenvolvedor conhecer essa linguagem. Existem várias bibliotecas para se trabalhar com JavaScript, por isso é importante ter conhecimento de algumas delas, pois é muito comum elas serem requisitos para o preenchimento de vagas JavaScript.
    </p>
    

    <p>
      Além da adequação ao mercado de trabalho, conhecer essas bibliotecas garante um aumento de produtividade e facilidade em tarefas comuns do dia a dia.
    </p>
    

    <p>
      Sendo assim, como todos os sites possuem algum tipo de interação utilizando o JavaScript, dificilmente você ficará fora do mercado.
    </p>
    

    <h2>
      Como aprender JavaScript?
    </h2>
    

    <p>
      Aqui no TreinaWeb temos toda uma trilha voltada para o JavaScript. Para você começar nessa linguagem, recomendamos o curso “JavaScript Básico”.
    </p>
    

    <p>
      Com ele, você aprenderá desde os conceitos básicos, fazendo com que você entenda toda a estrutura da linguagem.
    </p>
    

    <p>
      Aliando a teoria com a prática, você também irá desenvolver programas que irão interagir com os elementos de uma página web.
    </p>
    

    <h2>
      Conheça nossa metodologia
    </h2>
    

    <p>
      Nosso ambiente de aprendizagem é 100% online. Nossa metodologia de ensino é baseada em vídeo aulas, apostilas, exercícios aula-a-aula e exemplos completos. Você estuda no seu tempo e ao final pode baixar o seu certificado digital ou solicitar o certificado impresso.
    </p>
    

    <p>
      Na parte escrita, fornecemos um material totalmente atualizado e completo para que você possa usufruir do conteúdo, utilizando-se de textos e imagens. Já as vídeo-aulas são bem explicativas e práticas, abordando todo o conteúdo passo-a-passo, para que você entenda tudo da melhor maneira possível.
    </p>
    

    <p>
      Além disso, você encontrará exercícios para poder fixar, praticar e aplicar todo o conteúdo aprendido. São exercícios que são propostos em cada aula do curso. Eles podem variar entre alguns formatos, como por exemplo questões de escolha única, questões de múltipla escolha e complemento de trechos de código.
    </p>
    

    <p>
      Se ainda assim você tiver alguma dúvida, você poderá pedir ajuda aos nossos professores através de tickets de atendimento. Funciona como um canal direto entre você e o professor, onde você poderá realizar perguntas que serão respondidas muito brevemente!
    </p>
    

    <p>
      Além disso, ainda pelo sistema suporte, você também poderá consultar todas as perguntas que já foram respondidas a outros alunos! =)
    </p>
    

    <h2>
      Quais conhecimentos preciso ter para iniciar este curso?
    </h2>
    

    <p>
      O curso é recomendado para você iniciar seus estudos na linguagem JavaScript, no entanto, é interessante saber as bases necessárias para aprender qualquer linguagem de programação: “Lógica de Programação” e “Lógica de Programação Orientada a Objetos”.
    </p>
    

    <h2>
      Gostou do que viu?
    </h2>
    

    <p>
      Se entusiasmou com o JavaScript? Gostaria de conhecê-lo com mais detalhes? Sim, você pode dar uma espiadinha. Acesse o curso de JavaScript e veja algumas aulas de exemplo. :)
    </p>
    

    <p>
      Bons estudos!
    </p>
  </div>
</div>
`;
